repo/fsrepo/migrations/ipfsfetcher/ipfsfetcher.go

```diff
@@ -0,0 +1,332 @@
+package ipfsfetcher
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/ipfs/go-ipfs-config"
+	files "github.com/ipfs/go-ipfs-files"
+	"github.com/ipfs/go-ipfs/core"
+	"github.com/ipfs/go-ipfs/core/coreapi"
+	"github.com/ipfs/go-ipfs/core/node/libp2p"
+	"github.com/ipfs/go-ipfs/plugin/loader"
+	"github.com/ipfs/go-ipfs/repo/fsrepo"
+	"github.com/ipfs/go-ipfs/repo/fsrepo/migrations"
+	iface "github.com/ipfs/interface-go-ipfs-core"
+	"github.com/ipfs/interface-go-ipfs-core/options"
+	ipath "github.com/ipfs/interface-go-ipfs-core/path"
+	peer "github.com/libp2p/go-libp2p-core/peer"
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+const (
+	// Default maximum download size
+	defaultFetchLimit = 1024 * 1024 * 512
+
+	tempNodeTcpAddr = "/ip4/127.0.0.1/tcp/40010"
```

Comment:
Any reason this can't be `/ip4/127.0.0.1/tcp/0` we can then get the peer's "real" address by looking at the result of `ipfs.Swarm().LocalAddrs(ctx)`

Reply:
Did not know that was an option.  Nice.

Done.